### ISSUE TITLE
spkg-configure.m4 for gap

### ISSUE THREAD

**Issue by thierry-FreeBSD** [sagemath/sage#29644]
<div id="comment:0"></div>

This ticket adds an spkg-configure.m4 for gap, in order to use it from a system package if possible (see #27330).

With this, the installed gap is well detected, but for sagelib and gap packages to built, several paths must be modified:

- GAP_ROOT_DIR in src/sage/env.py

- GAP_ROOT in build/pkgs/gap_packages/spkg-install.in

- gap directory in src/sage/libs/gap/util.pyx

Moreover, the dependency on SAGE_LOCAL needs to be removed. Without it, one currently gets the following error

```
      File "/__w/sagetrac-mirror/sagetrac-mirror/src/sage/interfaces/expect.py", line 520, in _start
        raise RuntimeError("unable to start %s: %s" % (self.name(), msg))
    RuntimeError: unable to start gap: End Of File (EOF). Exception style platform.
    Gap finished running /__w/sagetrac-mirror/sagetrac-mirror/.tox/local-sudo-standard/local/bin/gap -r -b -p -T -E -o 401m -s 401m -m 64m /__w/sagetrac-mirror/sagetrac-mirror/src/sage/ext_data/gap/sage.g
    command: /__w/sagetrac-mirror/sagetrac-mirror/.tox/local-sudo-standard/local/bin/gap
    args: ['/__w/sagetrac-mirror/sagetrac-mirror/.tox/local-sudo-standard/local/bin/gap', '-r', '-b', '-p', '-T', '-E', '-o', '401m', '-s', '401m', '-m', '64m', '/__w/sagetrac-mirror/sagetrac-mirror/src/sage/ext_data/gap/sage.g']
    buffer (last 100 chars): b''
    before (last 100 chars): b'Set the environment variable SAGE_LOCAL.\r\n'
    after: <class 'pexpect.exceptions.EOF'>
    match: None
    match_index: None
    exitstatus: 1
    flag_eof: True
    pid: 3124
    child_fd: 26
    closed: False
    timeout: None
    delimiter: <class 'pexpect.exceptions.EOF'>
    logfile: None
    logfile_read: None
    logfile_send: None
    maxread: 4194304
    ignorecase: False
    searchwindowsize: None
    delaybeforesend: None
    delayafterclose: 0.1
    delayafterterminate: 0.1
    searcher: searcher_re:
        0: re.compile(b'gap> ')
```

CC:  @dimpase @orlitzky @slel @isuruf @antonio-rojas

Component: **build: configure**

Keywords: **gap, system packages**

Author: **Thierry Thomas, Samuel Lelièvre**

Branch/Commit: **[public/29644](https://github.com/sagemath/sagetrac-mirror/tree/public/29644) @ [`2a41f49`](https://github.com/sagemath/sagetrac-mirror/commit/2a41f491e5ce599ba185f5e93f02e0fcebd76c9e)**

_Issue created by migration from https://trac.sagemath.org/ticket/29644_

**Comment by tobiasdiez** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -7,3 +7,38 @@
 - GAP_ROOT in build/pkgs/gap_packages/spkg-install.in
 
 - gap directory in src/sage/libs/gap/util.pyx
+
+
+Moreover, the dependency on SAGE_LOCAL needs to be removed. Without it, one currently gets the following error
+
+```
+      File "/__w/sagetrac-mirror/sagetrac-mirror/src/sage/interfaces/expect.py", line 520, in _start
+        raise RuntimeError("unable to start %s: %s" % (self.name(), msg))
+    RuntimeError: unable to start gap: End Of File (EOF). Exception style platform.
+    Gap finished running /__w/sagetrac-mirror/sagetrac-mirror/.tox/local-sudo-standard/local/bin/gap -r -b -p -T -E -o 401m -s 401m -m 64m /__w/sagetrac-mirror/sagetrac-mirror/src/sage/ext_data/gap/sage.g
+    command: /__w/sagetrac-mirror/sagetrac-mirror/.tox/local-sudo-standard/local/bin/gap
+    args: ['/__w/sagetrac-mirror/sagetrac-mirror/.tox/local-sudo-standard/local/bin/gap', '-r', '-b', '-p', '-T', '-E', '-o', '401m', '-s', '401m', '-m', '64m', '/__w/sagetrac-mirror/sagetrac-mirror/src/sage/ext_data/gap/sage.g']
+    buffer (last 100 chars): b''
+    before (last 100 chars): b'Set the environment variable SAGE_LOCAL.\r\n'
+    after: <class 'pexpect.exceptions.EOF'>
+    match: None
+    match_index: None
+    exitstatus: 1
+    flag_eof: True
+    pid: 3124
+    child_fd: 26
+    closed: False
+    timeout: None
+    delimiter: <class 'pexpect.exceptions.EOF'>
+    logfile: None
+    logfile_read: None
+    logfile_send: None
+    maxread: 4194304
+    ignorecase: False
+    searchwindowsize: None
+    delaybeforesend: None
+    delayafterclose: 0.1
+    delayafterterminate: 0.1
+    searcher: searcher_re:
+        0: re.compile(b'gap> ')
+```
``````

**Comment by orlitzky** [comment]
Has anything improved in the past year and a half? @tobiasdiez is working on [a meson build for sagelib](https://github.com/sagemath/sage/pull/36524) that uses only

```meson
gap = cc.find_library('gap')
```

to find GAP. If it's that easy now, we could finally write the `spkg-configure.m4` for it; if it's not then it would be good to know the current list of pitfalls of that approach.

**Comment by mkoeppe** [comment]
Fixed in #36792
